### ISSUE TITLE
Few things to improve `tfa.losses.sigmoidal_focal_loss`

### ISSUE THREAD

**Issue by jeongukjae** [tensorflow/addons#2431]
**Describe the feature and the current behavior/state.**

1. Loss scales of `sigmoidal_focal_loss` and `tf.keras.losses.binary_crossentropy` are differ.

    `tfa.losses.sigmoidal_focal_loss` uses `binary_crossentropy` implemented in the TensorFlow, so I think it is reasonable to expect that focal loss and binary crossentropy will return the same values and actually they have same equation when `alpha` and `gamma` are `None`. However, `tfa.losses.sigmoidal_focal_loss` uses `tf.keras.backend.binary_crossentropy`, and does different reduction([focal loss uses reduce_sum](https://github.com/tensorflow/addons/blob/e2fcc148d129dad0c538ec688958997f207474aa/tensorflow_addons/losses/focal_loss.py#L142) and [binary crossentropy uses reduce_mean](https://github.com/tensorflow/tensorflow/blob/85c8b2a817f95a3e979ecd1ed95bff1dc1335cff/tensorflow/python/keras/losses.py#L1572-L1608)).
    I think using `reduce_mean` instead of `reduce_sum` or using `tf.keras.losses.binary_crossentropy` instead of `tf.keras.backend.binary_crossentropy` is better.

2. Comments for ignoring the `alpha` parameter.

    It looks like [ignoring the alpha parameter](https://github.com/tensorflow/addons/blob/e2fcc148d129dad0c538ec688958997f207474aa/tensorflow_addons/losses/focal_loss.py#L133-L135) is already implemented. It would be better if it is documented and type-hinted.

3. Wrong description for return values.

    https://github.com/tensorflow/addons/blob/e2fcc148d129dad0c538ec688958997f207474aa/tensorflow_addons/losses/focal_loss.py#L111-L112
    In comments of `sigmoidal_focal_loss`, I think the above lines should be like ``Weighted loss float `Tensor`.`` because the function returns not reduced values.

**Relevant information**
- Are you willing to contribute it (yes/no): Yes
*If you wish to contribute, then read the requirements for new contributions in [`CONTRIBUTING.md`](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#requirements-for-new-contributions-to-the-repository)*
- Are you willing to maintain it going forward? (yes/no):
- Is there a relevant academic paper? (if so, where): https://arxiv.org/pdf/1708.02002.pdf
- Does the relavent academic paper exceed 50 citations? (yes/no):
- Is there already an implementation in another framework? (if so, where):
- Was it part of tf.contrib? (if so, where):

**Which API type would this fall under (layer, metric, optimizer, etc.)**

**Who will benefit with this feature?**

**Any other info.**

**Comment by bhack** [comment]
Can you check the focal loss we have at https://github.com/tensorflow/models/blob/master/official/vision/keras_cv/losses/focal_loss.py?

**Comment by jeongukjae** [comment]
@bhack Sorry for the late reply. I will check that soon.

**Comment by jeongukjae** [comment]
@bhack I think that is a good implementation. It does not reduce losses like focal loss does in tfa, but the `alpha` parameter cannot be ignored. (However, ignoring alpha is a subtle problem because as the paper(https://arxiv.org/pdf/1708.02002.pdf) mentioned, alpha-balanced focal loss performs slightly better.)

**Comment by m-zheng** [comment]
@jeongukjae Hi, do you think the focal loss function implemented here is applicable to multi-classification? I think it's only for binary classification.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
